pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -92,27 +92,22 @@ func (lb *ListenerBuilder) buildWaypointInbound() []*listener.Listener {
 	//    forwarding to the VIP or Pod internal listener.
 	// 2. (many) VIP listeners, `inbound-vip||hostname|port`. This will apply service policies. For typical case (not redirecting to external service),
 	//    this will end up forwarding to a cluster for the same VIP, which will have endpoints for each Pod internal listener
-	// 3. (many) Pod listener, `inbound-pod||podip|port`. This is one per inbound pod. Will go through HCM if needed, in order to apply L7 policies (authz)
-	//    Note: we need both a pod listener and a VIP listener since we need to apply policies at different levels (routing vs authz).
-	// 4. Our final CONNECT listener, originating the tunnel
-	wls, svcs := FindAssociatedResources(lb.node, lb.push)
+	// 3. Our final CONNECT listener, originating the tunnel
+	_, svcs := FindAssociatedResources(lb.node, lb.push)
 
-	listeners = append(listeners, lb.buildWaypointInboundTerminateConnect(svcs, wls))
+	listeners = append(listeners, lb.buildWaypointInboundTerminateConnect(svcs))
 
 	// VIP listeners
 	listeners = append(listeners, lb.buildWaypointInboundVIP(svcs)...)
 
-	// Pod listeners
-	listeners = append(listeners, lb.buildWaypointInboundPod(wls)...)
-
 	listeners = append(listeners, lb.buildWaypointInboundOriginateConnect())
 
 	return listeners
 }
 
 // Our top level terminating CONNECT listener, `inbound TERMINATE`. This has a route per destination and decapsulates the CONNECT,
 // forwarding to the VIP or Pod internal listener.
```

Comment:
Does this comment (forward to pod internal listener) need to be updated?

Reply:
Updated I think.